app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php

```diff
@@ -82,6 +82,6 @@ public function canDisplayGiftmessage(): bool
         }
         /** @var Mage_GiftMessage_Helper_Message $helper */
         $helper = $this->helper('giftmessage/message');
-        return $helper->getIsMessagesAvailable('order', $this->getEntity(), $this->getEntity()->getStoreId());
+        return $helper->getIsMessagesAvailable('main', $this->getQuote(), $this->getStoreId());
```

Comment:
I think the first param `$type` refers to the type of entity, so
```suggestion
        return $helper->getIsMessagesAvailable('quote', $this->getQuote(), $this->getStoreId());
```

Reply:
@kiatng it was `main` before my MR, so 🤷‍♂️ 

https://github.com/OpenMage/magento-lts/pull/4266/files#diff-7d3121e3c6013546087c47238a88d86e0f15c12cbc344b8e893cbf1e94b19dc3L18

Reply:
https://github.com/OpenMage/magento-lts/blob/main/app/code/core/Mage/GiftMessage/Helper/Message.php#L95-L147

It seems it doesnt matter if its `main` or `quote`.

@Hanmac can you please check again to not have replaced any of the possible types? (items, item, order_item, or address_item),

Reply:
Added tests. LGTM.